src/allotropy/parsers/ctl_immunospot/ctl_immunospot_structure.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from dataclasses import dataclass
+import ntpath
+import re
+from typing import Optional
+
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.parsers.lines_reader import LinesReader
+from allotropy.parsers.utils.values import assert_not_none, try_float_or_none
+
+
+@dataclass
+class DeviceInfo:
+    device_identifier: str
+    model_number: str
+    equipment_serial_number: str
+    data_system_instance_id: str
+    basename: str
+    file_name: str
+    unc_path: str
+    software_name: str
+    software_version: str
+    measurement_time: str
+    analyst: str
+
+    @staticmethod
+    def create(reader: LinesReader) -> DeviceInfo:
+        data = {}
+        for i in range(3):  # for first three lines
+            line = assert_not_none(
+                reader.pop(),
+                msg=f"Unable to read line number {i+1}",
+            )
+            for element in line.split(";"):
+                parts = element.split(":", maxsplit=1)
+                if len(parts) > 1:
+                    data[parts[0].strip()] = parts[1].strip()
+
+        path = assert_not_none(
+            reader.pop(),
+            msg="Unable to read file path",
+        )
+
+        basename = ntpath.basename(path)
+        file_name, *_ = basename.split(".", maxsplit=1)
```

Comment:
Can we use pathlib instead of ntpath here?

Motivating example, you can re-write these two lines as:

file_name = Path(path).stem

Reply:
I tried to use path at first, but the path might be on posix or windows format.
With ntpath both cases are handled correctly.

Reply:
In this case, you can use `PureWindowsPath` from pathlib. I will support both Windows and Posix for this use case (parsing out filename)